// @datex2/typewriterjs/dist/core.js
!(function(e, t) {
  "object" == typeof exports && "object" == typeof module ? module.exports = t() : "function" == typeof define && define.amd ? define("Typewriter", [], t) : "object" == typeof exports ? exports.Typewriter = t() : e.Typewriter = t();
})("undefined" != typeof self ? self : this, (() => (() => {
  var e = { 3146: (e2, t2, n2) => {
    for (var r2 = n2(3491), o = "undefined" == typeof window ? n2.g : window, a = ["moz", "webkit"], i = "AnimationFrame", s = o["request" + i], u = o["cancel" + i] || o["cancelRequest" + i], l = 0; !s && l < a.length; l++) s = o[a[l] + "Request" + i], u = o[a[l] + "Cancel" + i] || o[a[l] + "CancelRequest" + i];
    if (!s || !u) {
      var c = 0, p = 0, d = [], f = 1e3 / 60;
      s = function(e3) {
        if (0 === d.length) {
          var t3 = r2(), n3 = Math.max(0, f - (t3 - c));
          c = n3 + t3, setTimeout((function() {
            var e4 = d.slice(0);
            d.length = 0;
            for (var t4 = 0; t4 < e4.length; t4++) if (!e4[t4].cancelled) try {
              e4[t4].callback(c);
            } catch (e5) {
              setTimeout((function() {
                throw e5;
              }), 0);
            }
          }), Math.round(n3));
        }
        return d.push({ handle: ++p, callback: e3, cancelled: false }), p;
      }, u = function(e3) {
        for (var t3 = 0; t3 < d.length; t3++) d[t3].handle === e3 && (d[t3].cancelled = true);
      };
    }
    e2.exports = function(e3) {
      return s.call(o, e3);
    }, e2.exports.cancel = function() {
      u.apply(o, arguments);
    }, e2.exports.polyfill = function(e3) {
      e3 || (e3 = o), e3.requestAnimationFrame = s, e3.cancelAnimationFrame = u;
    };
  }, 3491: function(e2) {
    (function() {
      var t2, n2, r2, o, a, i;
      "undefined" != typeof performance && null !== performance && performance.now ? e2.exports = function() {
        return performance.now();
      } : "undefined" != typeof process && null !== process && process.hrtime ? (e2.exports = function() {
        return (t2() - a) / 1e6;
      }, n2 = process.hrtime, o = (t2 = function() {
        var e3;
        return 1e9 * (e3 = n2())[0] + e3[1];
      })(), i = 1e9 * process.uptime(), a = o - i) : Date.now ? (e2.exports = function() {
        return Date.now() - r2;
      }, r2 = Date.now()) : (e2.exports = function() {
        return (/* @__PURE__ */ new Date()).getTime() - r2;
      }, r2 = (/* @__PURE__ */ new Date()).getTime());
    }).call(this);
  } }, t = {};
  function n(r2) {
    var o = t[r2];
    if (void 0 !== o) return o.exports;
    var a = t[r2] = { exports: {} };
    return e[r2].call(a.exports, a, a.exports, n), a.exports;
  }
  n.n = (e2) => {
    var t2 = e2 && e2.__esModule ? () => e2.default : () => e2;
    return n.d(t2, { a: t2 }), t2;
  }, n.d = (e2, t2) => {
    for (var r2 in t2) n.o(t2, r2) && !n.o(e2, r2) && Object.defineProperty(e2, r2, { enumerable: true, get: t2[r2] });
  }, n.g = (function() {
    if ("object" == typeof globalThis) return globalThis;
    try {
      return this || new Function("return this")();
    } catch (e2) {
      if ("object" == typeof window) return window;
    }
  })(), n.o = (e2, t2) => Object.prototype.hasOwnProperty.call(e2, t2);
  var r = {};
  return (() => {
    "use strict";
    n.d(r, { default: () => O });
    var e2 = n(3146), t2 = n.n(e2);
    const o = function(e3) {
      return new RegExp(/<[a-z][\s\S]*>/i).test(e3);
    }, a = function(e3, t3) {
      return Math.floor(Math.random() * (t3 - e3 + 1)) + e3;
    };
    var i = document.createElement("div");
    const s = function(e3) {
      if (e3 && "string" == typeof e3) {
        i.innerHTML = e3;
        var t3 = i.textContent || i.innerText || "";
        return i.textContent = "", t3;
      }
      return e3;
    };
    var u = "TYPE_CHARACTER", l = "REMOVE_CHARACTER", c = "REMOVE_ALL", p = "REMOVE_LAST_VISIBLE_NODE", d = "PAUSE_FOR", f = "CALL_FUNCTION", v = "ADD_HTML_TAG_ELEMENT", h = "CHANGE_DELETE_SPEED", m = "CHANGE_DELAY", y = "CHANGE_CURSOR", g = "PASTE_STRING", E = "HTML_TAG";
    function w(e3) {
      return w = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e4) {
        return typeof e4;
      } : function(e4) {
        return e4 && "function" == typeof Symbol && e4.constructor === Symbol && e4 !== Symbol.prototype ? "symbol" : typeof e4;
      }, w(e3);
    }
    function b(e3, t3) {
      var n2 = Object.keys(e3);
      if (Object.getOwnPropertySymbols) {
        var r2 = Object.getOwnPropertySymbols(e3);
        t3 && (r2 = r2.filter((function(t4) {
          return Object.getOwnPropertyDescriptor(e3, t4).enumerable;
        }))), n2.push.apply(n2, r2);
      }
      return n2;
    }
    function T(e3) {
      for (var t3 = 1; t3 < arguments.length; t3++) {
        var n2 = null != arguments[t3] ? arguments[t3] : {};
        t3 % 2 ? b(Object(n2), true).forEach((function(t4) {
          N(e3, t4, n2[t4]);
        })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e3, Object.getOwnPropertyDescriptors(n2)) : b(Object(n2)).forEach((function(t4) {
          Object.defineProperty(e3, t4, Object.getOwnPropertyDescriptor(n2, t4));
        }));
      }
      return e3;
    }
    function A(e3) {
      return (function(e4) {
        if (Array.isArray(e4)) return S(e4);
      })(e3) || (function(e4) {
        if ("undefined" != typeof Symbol && null != e4[Symbol.iterator] || null != e4["@@iterator"]) return Array.from(e4);
      })(e3) || (function(e4, t3) {
        if (e4) {
          if ("string" == typeof e4) return S(e4, t3);
          var n2 = {}.toString.call(e4).slice(8, -1);
          return "Object" === n2 && e4.constructor && (n2 = e4.constructor.name), "Map" === n2 || "Set" === n2 ? Array.from(e4) : "Arguments" === n2 || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n2) ? S(e4, t3) : void 0;
        }
      })(e3) || (function() {
        throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
      })();
    }
    function S(e3, t3) {
      (null == t3 || t3 > e3.length) && (t3 = e3.length);
      for (var n2 = 0, r2 = Array(t3); n2 < t3; n2++) r2[n2] = e3[n2];
      return r2;
    }
    function C(e3, t3) {
      for (var n2 = 0; n2 < t3.length; n2++) {
        var r2 = t3[n2];
        r2.enumerable = r2.enumerable || false, r2.configurable = true, "value" in r2 && (r2.writable = true), Object.defineProperty(e3, _(r2.key), r2);
      }
    }
    function N(e3, t3, n2) {
      return (t3 = _(t3)) in e3 ? Object.defineProperty(e3, t3, { value: n2, enumerable: true, configurable: true, writable: true }) : e3[t3] = n2, e3;
    }
    function _(e3) {
      var t3 = (function(e4) {
        if ("object" != w(e4) || !e4) return e4;
        var t4 = e4[Symbol.toPrimitive];
        if (void 0 !== t4) {
          var n2 = t4.call(e4, "string");
          if ("object" != w(n2)) return n2;
          throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return String(e4);
      })(e3);
      return "symbol" == w(t3) ? t3 : t3 + "";
    }
    const O = (function() {
      function n2(r3, i3) {
        var w2 = this;
        if ((function(e3, t3) {
          if (!(e3 instanceof t3)) throw new TypeError("Cannot call a class as a function");
        })(this, n2), N(this, "state", { cursorAnimation: null, lastFrameTime: null, pauseUntil: null, eventQueue: [], eventLoop: null, eventLoopPaused: false, reverseCalledEvents: [], calledEvents: [], visibleNodes: [], initialOptions: null, elements: { container: null, wrapper: document.createElement("span"), cursor: document.createElement("span") } }), N(this, "options", { strings: null, cursor: "|", delay: "natural", pauseFor: 1500, deleteSpeed: "natural", loop: false, autoStart: false, devMode: false, skipAddStyles: false, wrapperClassName: "Typewriter__wrapper", cursorClassName: "Typewriter__cursor", stringSplitter: null, onCreateTextNode: null, onRemoveNode: null }), N(this, "setupWrapperElement", (function() {
          w2.state.elements.container && (w2.state.elements.wrapper.className = w2.options.wrapperClassName, w2.state.elements.cursor.className = w2.options.cursorClassName, w2.state.elements.cursor.innerHTML = w2.options.cursor, w2.state.elements.container.innerHTML = "", w2.state.elements.container.appendChild(w2.state.elements.wrapper), w2.state.elements.container.appendChild(w2.state.elements.cursor));
        })), N(this, "start", (function() {
          return w2.state.eventLoopPaused = false, w2.runEventLoop(), w2;
        })), N(this, "pause", (function() {
          return w2.state.eventLoopPaused = true, w2;
        })), N(this, "stop", (function() {
          return w2.state.eventLoop && ((0, e2.cancel)(w2.state.eventLoop), w2.state.eventLoop = null), w2;
        })), N(this, "pauseFor", (function(e3) {
          return w2.addEventToQueue(d, { ms: e3 }), w2;
        })), N(this, "typeOutAllStrings", (function() {
          return "string" == typeof w2.options.strings ? (w2.typeString(w2.options.strings).pauseFor(w2.options.pauseFor), w2) : (w2.options.strings.forEach((function(e3) {
            w2.typeString(e3).pauseFor(w2.options.pauseFor).deleteAll(w2.options.deleteSpeed);
          })), w2);
        })), N(this, "typeString", (function(e3) {
          var t3 = arguments.length > 1 && void 0 !== arguments[1] ? arguments[1] : null;
          if (o(e3)) return w2.typeOutHTMLString(e3, t3);
          if (e3) {
            var n3 = (w2.options || {}).stringSplitter, r4 = s(e3), a2 = "function" == typeof n3 ? n3(r4) : r4.split("");
            w2.typeCharacters(a2, t3);
          }
          return w2;
        })), N(this, "pasteString", (function(e3) {
          var t3 = arguments.length > 1 && void 0 !== arguments[1] ? arguments[1] : null;
          if (o(e3)) return w2.typeOutHTMLString(e3, t3, true);
          if (e3) {
            var n3 = s(e3);
            w2.addEventToQueue(g, { character: n3, node: t3 });
          }
          return w2;
        })), N(this, "typeOutHTMLString", (function(e3) {
          var t3 = arguments.length > 1 && void 0 !== arguments[1] ? arguments[1] : null, n3 = arguments.length > 2 ? arguments[2] : void 0, r4 = (function(e4) {
            var t4 = document.createElement("div");
            return t4.innerHTML = e4, t4.childNodes;
          })(e3);
          if (r4.length > 0) for (var o2 = 0; o2 < r4.length; o2++) {
            var a2 = r4[o2], i4 = a2.innerHTML;
            a2 && 3 !== a2.nodeType ? (a2.innerHTML = "", w2.addEventToQueue(v, { node: a2, parentNode: t3 }), n3 ? w2.pasteString(i4, a2) : w2.typeString(i4, a2)) : a2.textContent && (n3 ? w2.pasteString(a2.textContent, t3) : w2.typeString(a2.textContent, t3));
          }
          return w2;
        })), N(this, "deleteAll", (function() {
          var e3 = arguments.length > 0 && void 0 !== arguments[0] ? arguments[0] : "natural";
          return w2.addEventToQueue(c, { speed: e3 }), w2;
        })), N(this, "changeDeleteSpeed", (function(e3) {
          if (!e3) throw new Error("Must provide new delete speed");
          return w2.addEventToQueue(h, { speed: e3 }), w2;
        })), N(this, "changeDelay", (function(e3) {
          if (!e3) throw new Error("Must provide new delay");
          return w2.addEventToQueue(m, { delay: e3 }), w2;
        })), N(this, "changeCursor", (function(e3) {
          if (!e3) throw new Error("Must provide new cursor");
          return w2.addEventToQueue(y, { cursor: e3 }), w2;
        })), N(this, "deleteChars", (function(e3) {
          if (!e3) throw new Error("Must provide amount of characters to delete");
          for (var t3 = 0; t3 < e3; t3++) w2.addEventToQueue(l);
          return w2;
        })), N(this, "callFunction", (function(e3, t3) {
          if (!e3 || "function" != typeof e3) throw new Error("Callback must be a function");
          return w2.addEventToQueue(f, { cb: e3, thisArg: t3 }), w2;
        })), N(this, "typeCharacters", (function(e3) {
          var t3 = arguments.length > 1 && void 0 !== arguments[1] ? arguments[1] : null;
          if (!e3 || !Array.isArray(e3)) throw new Error("Characters must be an array");
          return e3.forEach((function(e4) {
            w2.addEventToQueue(u, { character: e4, node: t3 });
          })), w2;
        })), N(this, "removeCharacters", (function(e3) {
          if (!e3 || !Array.isArray(e3)) throw new Error("Characters must be an array");
          return e3.forEach((function() {
            w2.addEventToQueue(l);
          })), w2;
        })), N(this, "addEventToQueue", (function(e3, t3) {
          var n3 = arguments.length > 2 && void 0 !== arguments[2] && arguments[2];
          return w2.addEventToStateProperty(e3, t3, n3, "eventQueue");
        })), N(this, "addReverseCalledEvent", (function(e3, t3) {
          var n3 = arguments.length > 2 && void 0 !== arguments[2] && arguments[2];
          return w2.options.loop ? w2.addEventToStateProperty(e3, t3, n3, "reverseCalledEvents") : w2;
        })), N(this, "addEventToStateProperty", (function(e3, t3) {
          var n3 = arguments.length > 2 && void 0 !== arguments[2] && arguments[2], r4 = arguments.length > 3 ? arguments[3] : void 0, o2 = { eventName: e3, eventArgs: t3 || {} };
          return w2.state[r4] = n3 ? [o2].concat(A(w2.state[r4])) : [].concat(A(w2.state[r4]), [o2]), w2;
        })), N(this, "runEventLoop", (function() {
          w2.state.lastFrameTime || (w2.state.lastFrameTime = Date.now());
          var e3 = Date.now(), n3 = e3 - w2.state.lastFrameTime;
          if (!w2.state.eventQueue.length) {
            if (!w2.options.loop) return;
            w2.state.eventQueue = A(w2.state.calledEvents), w2.state.calledEvents = [], w2.options = T({}, w2.state.initialOptions);
          }
          if (w2.state.eventLoop = t2()(w2.runEventLoop), !w2.state.eventLoopPaused) {
            if (w2.state.pauseUntil) {
              if (e3 < w2.state.pauseUntil) return;
              w2.state.pauseUntil = null;
            }
            var r4, o2 = A(w2.state.eventQueue), i4 = o2.shift();
            if (!(n3 <= (r4 = i4.eventName === p || i4.eventName === l ? "natural" === w2.options.deleteSpeed ? a(40, 80) : w2.options.deleteSpeed : "natural" === w2.options.delay ? a(120, 160) : w2.options.delay))) {
              var s2 = i4.eventName, b3 = i4.eventArgs;
              switch (w2.logInDevMode({ currentEvent: i4, state: w2.state, delay: r4 }), s2) {
                case g:
                case u:
                  var S2 = b3.character, C2 = b3.node, N2 = document.createTextNode(S2), _2 = N2;
                  w2.options.onCreateTextNode && "function" == typeof w2.options.onCreateTextNode && (_2 = w2.options.onCreateTextNode(S2, N2)), _2 && (C2 ? C2.appendChild(_2) : w2.state.elements.wrapper.appendChild(_2)), w2.state.visibleNodes = [].concat(A(w2.state.visibleNodes), [{ type: "TEXT_NODE", character: S2, node: _2 }]);
                  break;
                case l:
                  o2.unshift({ eventName: p, eventArgs: { removingCharacterNode: true } });
                  break;
                case d:
                  var O2 = i4.eventArgs.ms;
                  w2.state.pauseUntil = Date.now() + parseInt(O2);
                  break;
                case f:
                  var L = i4.eventArgs, x = L.cb, M = L.thisArg;
                  x.call(M, { elements: w2.state.elements });
                  break;
                case v:
                  var D = i4.eventArgs, P = D.node, j = D.parentNode;
                  j ? j.appendChild(P) : w2.state.elements.wrapper.appendChild(P), w2.state.visibleNodes = [].concat(A(w2.state.visibleNodes), [{ type: E, node: P, parentNode: j || w2.state.elements.wrapper }]);
                  break;
                case c:
                  var R = w2.state.visibleNodes, k = b3.speed, Q = [];
                  k && Q.push({ eventName: h, eventArgs: { speed: k, temp: true } });
                  for (var F = 0, H = R.length; F < H; F++) Q.push({ eventName: p, eventArgs: { removingCharacterNode: false } });
                  k && Q.push({ eventName: h, eventArgs: { speed: w2.options.deleteSpeed, temp: true } }), o2.unshift.apply(o2, Q);
                  break;
                case p:
                  var I = i4.eventArgs.removingCharacterNode;
                  if (w2.state.visibleNodes.length) {
                    var U = w2.state.visibleNodes.pop(), q = U.type, G = U.node, Y = U.character;
                    w2.options.onRemoveNode && "function" == typeof w2.options.onRemoveNode && w2.options.onRemoveNode({ node: G, character: Y }), G && G.parentNode.removeChild(G), q === E && I && o2.unshift({ eventName: p, eventArgs: {} });
                  }
                  break;
                case h:
                  w2.options.deleteSpeed = i4.eventArgs.speed;
                  break;
                case m:
                  w2.options.delay = i4.eventArgs.delay;
                  break;
                case y:
                  w2.options.cursor = i4.eventArgs.cursor, w2.state.elements.cursor.innerHTML = i4.eventArgs.cursor;
              }
              w2.options.loop && (i4.eventName === p || i4.eventArgs && i4.eventArgs.temp || (w2.state.calledEvents = [].concat(A(w2.state.calledEvents), [i4]))), w2.state.eventQueue = o2, w2.state.lastFrameTime = e3;
            }
          }
        })), r3) if ("string" == typeof r3) {
          var b2 = document.querySelector(r3);
          if (!b2) throw new Error("Could not find container element");
          this.state.elements.container = b2;
        } else this.state.elements.container = r3;
        i3 && (this.options = T(T({}, this.options), i3)), this.state.initialOptions = T({}, this.options), this.init();
      }
      var r2, i2;
      return r2 = n2, (i2 = [{ key: "init", value: function() {
        var e3, t3;
        this.setupWrapperElement(), this.addEventToQueue(y, { cursor: this.options.cursor }, true), this.addEventToQueue(c, null, true), !window || window.___TYPEWRITER_JS_STYLES_ADDED___ || this.options.skipAddStyles || (e3 = ".Typewriter__cursor{-webkit-animation:Typewriter-cursor 1s infinite;animation:Typewriter-cursor 1s infinite;margin-left:1px}@-webkit-keyframes Typewriter-cursor{0%{opacity:0}50%{opacity:1}100%{opacity:0}}@keyframes Typewriter-cursor{0%{opacity:0}50%{opacity:1}100%{opacity:0}}", (t3 = document.createElement("style")).appendChild(document.createTextNode(e3)), document.head.appendChild(t3), window.___TYPEWRITER_JS_STYLES_ADDED___ = true), true === this.options.autoStart && this.options.strings && this.typeOutAllStrings().start();
      } }, { key: "logInDevMode", value: function(e3) {
        this.options.devMode && console.log(e3);
      } }]) && C(r2.prototype, i2), Object.defineProperty(r2, "prototype", { writable: false }), n2;
    })();
  })(), r.default;
})()));
